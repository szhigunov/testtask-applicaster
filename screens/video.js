import React, {useState} from 'react';
import {VideoView} from '../components/videoview';
import {Preloader} from '../components/preloader';

export const VideoScreen = ({route}) => {
  const [loaded, setLoaded] = useState(false);
  let data = route.params.data;

  return (
    <>
      {!loaded && <Preloader title="Loading..." />}
      <VideoView
        onLoad={() => setLoaded(true)}
        onError={() => setLoaded(false)}
        uri={data.uri}
      />
    </>
  );
};
