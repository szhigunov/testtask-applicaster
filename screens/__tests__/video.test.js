/**
 * @format
 */

import 'react-native';
import React from 'react';
import {VideoScreen} from '../video';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('VideoScreen renders correctly', () => {
  let root = renderer.create(<VideoScreen route={{params: {data: {uri: 'someurl'}}}} />);
  expect(root.toJSON()).toMatchSnapshot();
});
