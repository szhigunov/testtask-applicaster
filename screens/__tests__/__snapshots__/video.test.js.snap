// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoScreen renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View>
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
      <Text
        style={
          Object {
            "paddingTop": 10,
          }
        }
      >
        Loading...
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Video
      controls={true}
      muted={false}
      onError={[Function]}
      onLoad={[Function]}
      paused={false}
      playInBackground={false}
      playWhenInactive={false}
      progressUpdateInterval={250}
      rate={1}
      repeat={false}
      resizeMode="contain"
      source={
        Object {
          "uri": "someurl",
        }
      }
      style={
        Object {
          "flex": 1,
          "width": "100%",
        }
      }
      volume={0}
    />
  </View>,
]
`;
